common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -6322,29 +6322,67 @@ public ZooKeeperHiveHelper getZKConfig() {
       .trustStorePassword(trustStorePassword).build();
   }
 
+  public static HiveConf create() {
+    return new HiveConf();
+  }
+
+  public static HiveConf create(Class<?> cls) {
+    return new HiveConf(cls);
+  }
+
+  public static HiveConf create(Configuration other, Class<?> cls) {
+    return new HiveConf(other, cls);
+  }
+
+
+  public static HiveConf create(HiveConf other) {
+    return new HiveConf(other);
+  }
+
+  /**
+   * Instantiating HiveConf is deprecated. Please use
+   * HiveConf#create() to construct a Configuration,
+   * this method will become private eventually.
+   * @deprecated Please use create method instead.
+   */
   public HiveConf() {
```

Comment:
Please include the `@Deprecated` annotation it is useful for some purposes.

"Instantiating HiveConf is deprecated" phrase is redundant. IDEs and compilers will generate automatically similar warnings when calling this method.

"Instatiating...eventually" the whole paragraph concerns deprecation so it should be associated with the `@deprecated` tag.

Based on [1] and the comments above, I would suggest the following:
```java
  /**
   * @deprecated This method will become private eventually; Use {@link #create()} instead. 
   */
  @Deprecated
  public HiveConf() {
```

[1] https://docs.oracle.com/javase/7/docs/technotes/guides/javadoc/deprecation/deprecation.html

Reply:
agreed